ten-simple-rules-dockerfiles.Rmd

```diff
@@ -473,6 +500,9 @@ With plain Docker, you can override the defaults as part of the `docker run` com
 In any case you should document different variants, if you choose to provide them, in a `Makefile` [@wikipedia_contributors_make_2019].
 To support both one click execution and interactive interfaces and even allow for custom configuration, it's helpful to expose settings via a configuration file which can be bound from the host, via environment variables [@knoth_reproducibility_2017], or special Docker-based wrappers such as Kliko [@molenaar_klikoscientific_2018].
 
+
+[SJE: I think I get the two style of working -- batch and interactive might be my way of thinking about them.  But it feels a bit clunky above and certainly too detailed.  might be better served with two example dockerfiles: once doing a batch analysis, and the other whose goal is to allow some interaction.]
+
```

Comment:
I think it's a bit contradictory to talk about reproducible execution and then advocate for any graphical user interface. If someone brings a sys admin a python notebook someone will undoubtably roll in their grave.

Reply:
I like the example though of a docker container that generates a pdf (from a knitr doucument) but then also allows the user to run rstudio to further edit the knitr doc.

Reply:
ah I understand what you mean - yes that's a very good point! Just don't run a rockr rstudio container that is converted to singularity, results in a very bad outcome :P